jormungandr-lib/src/interfaces/mint_token.rs

```diff
@@ -0,0 +1,93 @@
+use super::Value;
+use crate::crypto::account::Identifier;
+use chain_impl_mockchain::{
+    certificate,
+    tokens::{minting_policy, name},
+};
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+struct TokenName(name::TokenName);
+
+impl From<name::TokenName> for TokenName {
+    fn from(val: name::TokenName) -> Self {
+        Self(val)
+    }
+}
+
+impl Serialize for TokenName {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        hex::encode(self.0.as_ref()).serialize(serializer)
```

Comment:
I would add different behavior depending on whether the serializer/deserializer is human readable or not. There's no reason to dump this in hex format in the file is binary

Reply:
Good point !
Have updated it